clevertap-core/src/main/java/com/clevertap/android/sdk/CleverTapAPI.java

```diff
@@ -3515,35 +3510,83 @@ public void clearInAppResources(boolean expiredOnly) {
-        LegacyInAppStore legacyInAppStore = storeRegistry.getLegacyInAppStore();
-
-        if (inAppAssetStore == null || legacyInAppStore == null) {
-            logger.info("There was a problem clearing resources because instance is not completely initialised, please try again after some time");
+        FileResourcesRepoImpl impl = FileResourcesRepoFactory.createFileResourcesRepo(context, logger, storeRegistry);
+        if (impl == null) {
+            logger.info(
+                    "There was a problem clearing resources because instance is not completely initialised, please try again after some time");
             return;
         }
 
-        InAppResourceProvider inAppResourceProvider = new InAppResourceProvider(context, logger);
-        InAppCleanupStrategy cleanupStrategy = new InAppCleanupStrategyExecutors(inAppResourceProvider);
-        InAppImagePreloaderStrategy preloadStrategy = new InAppImagePreloaderExecutors(
-                inAppResourceProvider,
-                logger
-        );
-
-        InAppImageRepoImpl impl = new InAppImageRepoImpl(
-                cleanupStrategy,
-                preloadStrategy,
-                inAppAssetStore,
-                legacyInAppStore
-        );
         if (expiredOnly) {
             impl.cleanupStaleImagesNow();
         } else {
             impl.cleanupAllImages();
         }
     }
+
+    /**
+     * Deletes all types of files which are preloaded for SDK features like custom in-app templates, app functions and
+     * variables etc.
+     *
+     * @param expiredOnly to clear only files which will not be needed further for SDK features like custom in-app
+     *                    templates, app functions and variables etc.
+     */
+    public void clearFileResources(boolean expiredOnly) {
+
+        Logger logger = coreState.getConfig().getLogger();
+
+        StoreRegistry storeRegistry = coreState.getStoreRegistry();
+        if (storeRegistry == null) {
+            logger.info(
+                    "There was a problem clearing file resources because instance is not completely initialised, please try again after some time");
+            return;
+        }
+
+        FileResourcesRepoImpl impl = FileResourcesRepoFactory.createFileResourcesRepo(context, logger, storeRegistry);
+        if (impl == null) {
+            logger.info(
+                    "There was a problem clearing file resources because instance is not completely initialised, please try again after some time");
+            return;
+        }
+
+        if (expiredOnly) {
+            impl.cleanupStaleFilesNow();
+        } else {
+            impl.cleanupAllFiles();
+        }
+    }
+
+    /**
+     * Checks if a file exists for the given non-null url.
+     *
+     * @param url the non-null url for which to check the existence of the file.
+     * @return true if a file exists for the specified url, false otherwise.
+     */
+    public boolean isFileExistsForUrl(@NonNull String url){
```

Comment:
this can be renamed to `checkFileForUrl` or `doesFileExistForUrl`

Reply:
Fixed in [49f583](https://github.com/CleverTap/clevertap-android-sdk/commit/49f583b5f6f025a8ca3d1f0a70a612b3166f0822)